### PR TITLE
Default transition targetstate to transition configName

### PR DESCRIPTION
**Pull request by PiZZAD0X**
**When merged this pull request will:**
- Change the default transition targetstate value to the transition configName, so that most transitions between states can have their transition state and targetstate defined without a near duplicate entry.

Example: 

```
class Init {
    onStateEntered = QFUNC(onCacheInit);
    class Cache {
        targetState = "Cache";
        events[] = {QGVAR(cache)};
    };

    class Uncache {
        targetState = "Uncache";
        events[] = {QGVAR(uncache)};
    };
};
```

Could now be written as: 

```
class Init {
    onStateEntered = QFUNC(onCacheInit);
    class Cache {
        events[] = {QGVAR(cache)};
    };

    class Uncache {
        events[] = {QGVAR(uncache)};
    };
};
```

while targetState 's that deviate from the transition configName are supported.

### CODE REVIEW

File: addons/statemachine/fnc_createFromConfig.sqf [hunk 1]
```diff
@@ -45,7 +45,7 @@ private _stateMachine = [_list, _skipNull] call FUNC(create);
     private _state = configName _x;
     {
         private _transition = configName _x;
-        private _targetState = getText (_x >> "targetState");
+        private _targetState = [_x,"targetState",_transition] call BIS_fnc_returnConfigEntry;
```
Comment (veteran29) [inline addons/statemachine/fnc_createFromConfig.sqf]:
```suggestion
        private _targetState = [_x >> "targetState", "text", _transition] call CBA_fnc_getConfigEntry;
```

Comment (PiZZAD0X) [reply addons/statemachine/fnc_createFromConfig.sqf]: This or is/getText method?

Comment (commy2) [reply addons/statemachine/fnc_createFromConfig.sqf]: isText

**Comment by BaerMitUmlaut** [comment]
Sure why not.

**Pull request merged**